substrate/frame/collective/src/lib.rs

```diff
@@ -269,6 +418,12 @@ pub mod pallet {
 	pub type ProposalOf<T: Config<I>, I: 'static = ()> =
 		StorageMap<_, Identity, T::Hash, <T as Config<I>>::Proposal, OptionQuery>;
 
+	/// Consideration cost created for publishing and storing a proposal. Determined by
+	/// [Config::Consideration] and may not be applicable for certain proposals.
+	#[pallet::storage]
+	pub type CostOf<T: Config<I>, I: 'static = ()> =
+		StorageMap<_, Identity, T::Hash, (T::AccountId, T::Consideration), OptionQuery>;
```

Comment:
Right now they have to code it up, same as with all dynamic deposits. 
Or it can be provided by some runtime api, and I think it should be done on a runtime crate level.

Reply:
On the pallet level we only know that there might be some consideration/cost based on proposal count, what is that cost we do not know, it can be some balance that will be put on hold, but also can be something else. 
Solution can be that we add an additional bound to `Consideration: Convert<ProposalCount, Config::Cost>`, plus we add `Cost` type as type parameter to the config. But I think better users provide a runtime api call themself if they want to.

Reply:
> Or it can be provided by some runtime api, and I think it should be done on a runtime crate level.

Yes, and ofc this runtime api is best effort, as in it might be different between the time of invocation and when a proposal is actually created. All non-core APIs are stateless, so you can just call into the same Consideration in the API, and track the side effects. 

Or better, perhaps we can think of a nicer API for `Consideration`s to return their side effect, so it can be predicted. 

In any case, I hope we can exclude "let UIs reverse engineer it" from our options :)

Reply:
I could actually provide a solution without additional bounds and type parameters to config. But I cannot declare an runtime api for a pallet that might have multiple instances. So I think we can leave it for a runtime engineer.

Reply:
I cannot declare the runtime API that will work for multiple pallet instances